src/auto-resolve.ts

```diff
@@ -0,0 +1,136 @@
+import fs from 'fs';
+import path from 'path';
+import {parse} from '@babel/parser';
+import traverse, {NodePath} from '@babel/traverse';
+import {
+  CallExpression,
+  ImportDeclaration,
+  isImportDeclaration,
+  isCallExpression,
+} from '@babel/types';
+
+interface DependencyTree {
+  [key: string]: string[];
+}
+
+/**
+ * Return module name from static import
+ */
+function importModuleName(node: NodePath<ImportDeclaration>) {
```

Comment:
I know we talked about this offline briefly, but just to confirm: this PR installs the entire package of every imported package, correct? it doesn't tree-shake and remove unused imports/exports within installed dependencies?

I'm only asking because I'm really excited about that feature, but I want to make sure that it's talked about first and only implemented as production-only behind the `--optimize` flag (since doing that during dev would turn @pika/web into a much heavier, re-run-after-every-save kind of tool)

Reply:
As a first pass, it taps into the Rollup thing that’s already happening. Another way to look at it is it auto-populates the `webDependencies` list.

What you’re talking about is definitely possible, but that’d happen at the Rollup layer, I think?

Reply:
Yea exactly, it would need to be a custom plugin. I asked the rollup team for feedback on it a few months ago and got some good advice, I'll pull that up and find it if you're curious.

Reply:
Yes! Definitely curious about this. This should be a Rollup plugin one way or the other (surprised it’s not TBH)